parse/src/lockfiles/mod.rs

```diff
@@ -0,0 +1,185 @@
+use std::fmt::Display;
+use std::path::Path;
+use std::str::FromStr;
+
+pub use csharp::CSProj;
+pub use java::{GradleLock, Pom};
+pub use javascript::{PackageLock, YarnLock};
+use phylum_types::types::package::{PackageDescriptor, PackageType};
+pub use python::{PipFile, Poetry, PyRequirements};
+pub use ruby::GemLock;
+use serde::de::IntoDeserializer;
+use serde::{Deserialize, Serialize};
+use strum::{EnumIter, IntoEnumIterator};
+
+mod csharp;
+mod java;
+mod javascript;
+mod parsers;
+mod python;
+mod ruby;
+
+/// A file format that can be parsed.
+#[derive(
+    Clone, Copy, Debug, Deserialize, EnumIter, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize,
```

Comment:
I'm personally not a big fan of pulling in dependencies just to implement iteration on enums.